controllers/machine_complex_heuristic_actor.py

```diff
@@ -11,6 +13,57 @@ class MachineComplexHeuristicActor(MachineABPruningActor):
     def __init__(self, name: str, state, playerId):
         super().__init__(name, state, playerId)
 
-    def __get_heuristic(self, game):
+    def _get_heuristic(self, game):
         # TODO: implement heuristic!!!
-        pass
+
+        # The more pawns one has compared to the number of pawns
+        # the opponent has, the better.
+        count_delta = self._get_fields_delta(game._get_bord(),
+                                             game._current_player)
+
+        # The further advanced a pawn, the better.
+        adv_board = deepcopy(game._get_bord())
+        size = game._size
+
+        for r in range(size):
+            for c in range(size):
+                if adv_board[r][c] == Player.OPP:
+                    # best row is at the "bottom" of the board
+                    # -1 to prevent the first move to always be two steps.
+                    # squared weight
+                    adv_board[r][c] *= (r - 1)**2
+                elif adv_board[r][c] == Player.USER:
+                    # best row is at the "top" of the board
+                    adv_board[r][c] *= (size - 1 - r - 1)**2
+
+        adv_delta = self._get_fields_delta(adv_board, game._current_player)
+
+        # A pawn that has an unblocked path to the finish line is great
+        board = game._get_bord()

```

Comment:
use adv_board for this to get increased score the further a pawn has progressed especially if his sight is free.

Reply:
done